osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -148,50 +152,32 @@ QueryData genNtfsAclPerms(QueryContext& context) {
                                         nullptr,
                                         nullptr);
     if (ERROR_SUCCESS != result) {
-      VLOG(1) << "GetExplicitEnteriesFromAcl Error " << result;
-      continue;
-    }
-
-    // get list of ACEs from DACL pointer
-    unsigned long aceCount = 0;
-    PEXPLICIT_ACCESSW aceList = nullptr;
-    result = GetExplicitEntriesFromAclW(dacl, &aceCount, &aceList);
-    if (ERROR_SUCCESS != result) {
-      VLOG(1) << "GetExplicitEnteriesFromAcl Error " << result;
+      VLOG(1) << "GetNamedSecurityInfo Error " << result;
       continue;
     }
 
-    // Loop through list of entries
-    auto aceItem = aceList;
-    for (unsigned long aceIndex = 0; aceIndex < aceCount;
-         aceItem++, aceIndex++) {
+    ACCESS_ALLOWED_ACE* pAce = NULL;
+    // Loop through the ACEs and display the information.
+    for (int cAce = 0; cAce < dacl->AceCount; cAce++) {
       Row r;
-
-      auto perms = accessPermsToStr(aceItem->grfAccessPermissions);
-
-      // TODO(6129): Determine the best way to report unknown values.
-      // Investigate the documentation for correct grfInheritance usage.
-      auto accessModeIter = kAccessModeToStr.find(aceItem->grfAccessMode);
-      auto accessMode = (accessModeIter != kAccessModeToStr.end())
-                            ? accessModeIter->second
-                            : "Unknown";
-      auto interitedFromiter = kInheritanceToStr.find(aceItem->grfInheritance);
-      auto inheritedFrom = (interitedFromiter != kInheritanceToStr.end())
-                               ? interitedFromiter->second
-                               : "Unknown";
-      auto trusteeName = trusteeToStr(aceItem->Trustee);
+      // Get ACE
+      if (GetAce(dacl, cAce, (LPVOID*)&pAce) == FALSE) {
+        VLOG(1) << "GetAce failed. Error = " << GetLastError();
```

Comment:
nitpick, there's inconsistencies in how the verbose logging is happening, see above on line 155. I'd recommend stick to the precedent for now and use `"GetAce Error" << GetLastError()`.